packages/@lwc/ssr-compiler/src/compile-template/transformers/element.ts

```diff
@@ -41,17 +42,31 @@ const bYield = (expr: EsExpression) => b.expressionStatement(b.yieldExpression(e
 // TODO [#4714]: scope token renders as a suffix for literals, but prefix for expressions
 const bConditionalLiveYield = esTemplateWithYield`
     {
-        const shouldRenderScopeToken = ${/* isClass */ is.literal} &&
+        const attrName = ${/* attribute name */ is.literal};
+        let attrValue = ${/* attribute value expression */ is.expression};
+        const isHtmlBooleanAttr = ${/* isHtmlBooleanAttr */ is.literal};
+
+        const shouldRenderScopeToken = attrName === 'class' &&
```

Comment:
There is no need to precompute `attrName === 'class'`. Let the minifier/Rollup figure out the tree-shaking.